### PR TITLE
Simplify string length checks

### PR DESCRIPTION
**Pull request by Juiceman**
We are looking for empty values, so use .isEmpty()

### CODE REVIEW

File: src/freenet/clients/http/bookmark/Bookmark.java [hunk 1]
```diff
@@ -18,7 +18,7 @@ public String getVisibleName() {
 	}
 
 	protected void setName(String s) {
-		name = (s.length() > 0 ? s : NodeL10n.getBase().getString("Bookmark.noName"));
+		name = (!s.isEmpty() ? s : NodeL10n.getBase().getString("Bookmark.noName"));
```
Comment (Bombe) [inline src/freenet/clients/http/bookmark/Bookmark.java:21]: Another case of negation in the ternary condition… again, I’m not sure we _need_ to change this…

File: src/freenet/node/NewPacketFormat.java [hunk 2]
```diff
@@ -169,7 +169,7 @@ List<byte[]> handleDecryptedPacket(NPFPacket packet, SessionKey sessionKey) {
 		
 		boolean dontAck = false;
 		boolean wakeUp = false;
-		if(packet.getError() || (packet.getFragments().size() == 0)) {
+		if(packet.getError() || (packet.getFragments().isEmpty())) {
```
Comment (Bombe) [inline src/freenet/node/NewPacketFormat.java:172]: Parens.

File: src/freenet/pluginmanager/PluginManager.java [hunk 3]
```diff
@@ -1001,7 +1001,7 @@ public List<OfficialPluginDescription> findAvailablePlugins() {
 	}
 
 	public OfficialPluginDescription isOfficialPlugin(String name) {
-		if((name == null) || (name.trim().length() == 0))
+		if((name == null) || (name.trim().isEmpty()))
```
Comment (Bombe) [inline src/freenet/pluginmanager/PluginManager.java:1004]: Parens.

File: src/freenet/support/FileLoggerHook.java [hunk 4]
```diff
@@ -636,7 +636,7 @@ public void findOldLogFiles(GregorianCalendar gc) {
 					name = name.substring(0, name.length()-".log.gz".length());
 				}
 				name = name.substring(prefix.length());
-				if((name.length() == 0) || (name.charAt(0) != '-')) {
+				if((name.isEmpty()) || (name.charAt(0) != '-')) {
```
Comment (Bombe) [inline src/freenet/support/FileLoggerHook.java:639]: Parens, below as well.

File: src/freenet/support/SimpleFieldSet.java [hunk 5]
```diff
@@ -201,7 +201,7 @@ private void read(LineReader br, int maxLength, int bufferSize, boolean utfOrIso
 				Logger.error(this, "No end marker");
 				break;
 			}
-			if ((line.length() == 0)) continue; // ignore
+			if ((line.isEmpty())) continue; // ignore
```
Comment (Bombe) [inline src/freenet/support/SimpleFieldSet.java:204]: So many parens here! 😀

**Review by Bombe — CHANGES_REQUESTED** [review]

File: src/freenet/client/ClientMetadata.java [hunk 6]
```diff
@@ -54,7 +54,7 @@ public static ClientMetadata construct(DataInputStream dis) throws MetadataParse
 	/** Get the document MIME type. Will always be a valid MIME type, unless there
 	 * has been an error; if it is unknown, will return application/octet-stream. */
 	public String getMIMEType() {
-		if((mimeType == null) || (mimeType.length() == 0))
+		if((mimeType == null) || (mimeType.isEmpty()))
```
[the remark this answers is not in the document]
Comment (Juiceman) [reply src/freenet/client/ClientMetadata.java]: Fixed in ed98d8c127320b71b507f1524cce60e0d5efa356

File: src/freenet/client/filter/HTMLFilter.java [hunk 7]
```diff
@@ -2647,7 +2647,7 @@ Map<String, Object> sanitizeHash(Map<String, Object> h,
 			if (type != null) {
 				String[] typesplit = splitType(type);
 				type = typesplit[0];
-				if ((typesplit[1] != null) && (typesplit[1].length() > 0))
+				if ((typesplit[1] != null) && (!typesplit[1].isEmpty()))
```
[the remark this answers is not in the document]
Comment (Juiceman) [reply src/freenet/client/filter/HTMLFilter.java]: Fixed in ed98d8c127320b71b507f1524cce60e0d5efa356

File: src/freenet/client/filter/HTMLFilter.java [hunk 8]
```diff
@@ -3454,7 +3454,7 @@ static String sanitizeURI(
 		if(logMINOR)
 			Logger.minor(HTMLFilter.class, "Sanitizing URI: "+suri+" ( override type "+overrideType +" override charset "+overrideCharset+" ) inline="+inline, new Exception("debug"));
 		boolean addMaybe = false;
-		if((overrideCharset != null) && (overrideCharset.length() > 0))
+		if((overrideCharset != null) && (!overrideCharset.isEmpty()))
```
[the remark this answers is not in the document]
Comment (Juiceman) [reply src/freenet/client/filter/HTMLFilter.java]: Fixed in ed98d8c127320b71b507f1524cce60e0d5efa356

File: src/freenet/clients/fcp/FCPServer.java [hunk 9]
```diff
@@ -909,7 +909,7 @@ public void makePersistentGlobalRequest(FreenetURI fetchURI, boolean filterData,
 
 	private File makeReturnFilename(FreenetURI uri, String expectedMimeType, File downloadsDir) {
 		String ext;
-		if((expectedMimeType != null) && (expectedMimeType.length() > 0) &&
+		if((expectedMimeType != null) && (!expectedMimeType.isEmpty()) &&
```
[the remark this answers is not in the document]
Comment (Juiceman) [reply src/freenet/clients/fcp/FCPServer.java]: Fixed in ed98d8c127320b71b507f1524cce60e0d5efa356

File: src/freenet/support/SimpleFieldSet.java (hunk 5 above)
Comment (Juiceman) [reply src/freenet/support/SimpleFieldSet.java]: Fixed in ed98d8c127320b71b507f1524cce60e0d5efa356

File: src/freenet/pluginmanager/PluginManager.java (hunk 3 above)
Comment (Juiceman) [reply src/freenet/pluginmanager/PluginManager.java]: Fixed in ed98d8c127320b71b507f1524cce60e0d5efa356

File: src/freenet/clients/http/QueueToadlet.java [hunk 10]
```diff
@@ -229,7 +229,7 @@ public void handleMethodPOST(URI uri, HTTPRequest request, final ToadletContext
 				}
 			}
 
-			if(request.isPartSet("delete_request") && (request.getPartAsStringFailsafe("delete_request", 128).length() > 0)) {
+			if(request.isPartSet("delete_request") && (!request.getPartAsStringFailsafe("delete_request", 128).isEmpty())) {
```
[the remark this answers is not in the document]
Comment (Juiceman) [reply src/freenet/clients/http/QueueToadlet.java]: Fixed in ed98d8c127320b71b507f1524cce60e0d5efa356

File: src/freenet/clients/http/HTTPRequestImpl.java [hunk 11]
```diff
@@ -123,7 +123,7 @@ public HTTPRequestImpl(String path, String encodedQueryString, String method) th
 		this.data = null;
 		this.parts = null;
 		this.bucketfactory = null;
-		if ((encodedQueryString!=null) && (encodedQueryString.length()>0)) {
+		if ((encodedQueryString!=null) && (!encodedQueryString.isEmpty())) {
```
[the remark this answers is not in the document]
Comment (Juiceman) [reply src/freenet/clients/http/HTTPRequestImpl.java]: Fixed in ed98d8c127320b71b507f1524cce60e0d5efa356

File: src/freenet/support/FileLoggerHook.java (hunk 4 above)
Comment (Juiceman) [reply src/freenet/support/FileLoggerHook.java]: Fixed in ed98d8c127320b71b507f1524cce60e0d5efa356

File: src/freenet/node/NewPacketFormat.java (hunk 2 above)
Comment (Juiceman) [reply src/freenet/node/NewPacketFormat.java]: Fixed in ed98d8c127320b71b507f1524cce60e0d5efa356

File: src/freenet/clients/fcp/LoadPlugin.java [hunk 12]
```diff
@@ -38,7 +38,7 @@ public LoadPlugin(SimpleFieldSet fs) throws MessageInvalidException {
 		if(pluginURL == null)
 			throw new MessageInvalidException(ProtocolErrorMessage.MISSING_FIELD, "Must contain a PluginURL field", identifier, false);
 		String type = fs.get("URLType");
-		if ((type != null) && (type.trim().length() > 0))
+		if ((type != null) && (!type.trim().isEmpty()))
```
[the remark this answers is not in the document]
Comment (Juiceman) [reply src/freenet/clients/fcp/LoadPlugin.java]: Fixed in ed98d8c127320b71b507f1524cce60e0d5efa356

File: src/freenet/clients/http/bookmark/Bookmark.java (hunk 1 above)
Comment (Juiceman) [reply src/freenet/clients/http/bookmark/Bookmark.java]: Reverted.  Unnecessarily confusing.

File: src/freenet/clients/http/FirstTimeWizardNewToadlet.java [hunk 13]
```diff
@@ -266,17 +266,17 @@ private void detectBandwidthLimit() {
 
         private Map<String, Object> toModel() {
             HashMap<String, Object> model = new HashMap<>();
-            model.put("knowSomeone", knowSomeone.length() > 0 ? "checked" : "");
-            model.put("connectToStrangers", connectToStrangers.length() > 0 ? "checked" : "");
-            model.put("haveMonthlyLimit", haveMonthlyLimit.length() > 0 ? "checked" : "");
+            model.put("knowSomeone", !knowSomeone.isEmpty() ? "checked" : "");
+            model.put("connectToStrangers", !connectToStrangers.isEmpty() ? "checked" : "");
+            model.put("haveMonthlyLimit", !haveMonthlyLimit.isEmpty() ? "checked" : "");
```
[the remark this answers is not in the document]
Comment (Juiceman) [reply src/freenet/clients/http/FirstTimeWizardNewToadlet.java]: Reverted.  Unnecessarily confusing.

[the remark this answers is not in the document]
Comment (Juiceman) [reply src/freenet/clients/http/FirstTimeWizardNewToadlet.java]: Thank you for your reviews!

**Review by Bombe — APPROVED** [review]

**Comment by ArneBab** [comment]
merged — thank you!

**Pull request merged**